### PR TITLE
Organize luthier

### PR DESCRIPTION
**Pull request by gab-umich**
Attempting to re-organize the Java and Lua code in the Luthier design module.

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: luthier/scripts/emptyLuthierApp.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+# prepares json files
+cd $DIR/../src/main/lua/
+lua config.lua
+# this is equivalent to lua config.lua app ../../../target/classes/
+
+# install fili using mvn
+cd $DIR/../../
+mvn install -DskipTests -Dcheckstyle.skip
+# runs luthier wiki example on port 9012
```
Comment (michael-mclawhorn) [inline luthier/scripts/emptyLuthierApp.sh]: Remove reference to wiki

**Comment by gab-umich** [comment]
I don't think a `final char underscore = '_';` is a bad style. I will recommend ignoring this unless we have a good reason.

**Comment by gab-umich** [comment]
Also, according to https://github.com/koalaman/shellcheck/wiki/SC2086, we should use `shell_script arg1 "$DIR"` instead of `shell_script arg1 $DIR`. Although the later one is valid, it might not be robust enough to handle word splitting. It should not be a problem for DIR specifically, but we should use this to encourage our user to be aware of this case.

File: luthier/scripts/emptyLuthierApp.sh (hunk 1 above)
Comment (gab-umich) [reply luthier/scripts/emptyLuthierApp.sh]: resolved

**Pull request merged**